src/NServiceBus.Core/Transports/Msmq/QueueCreator.cs

```diff
@@ -31,83 +31,59 @@ public Task CreateQueueIfNecessary(QueueBindings queueBindings, string identity)
 
         void CreateQueueIfNecessary(string address, string identity)
         {
-            if (address == null)
+            Guard.AgainstNullAndEmpty(nameof(address), address);
+
+            var msmqAddress = MsmqAddress.Parse(address);
+            
+            Logger.Debug($"Trying to open queue '{address}'.");
+
+            MessageQueue queue;
+            if (!MsmqUtilities.TryOpenQueue(msmqAddress, out queue))
             {
-                return;
+                Logger.Warn($"Queue '{address}' does not exist.");
+                Logger.Debug($"Creating queue: {address}");
+
+                queue = CreateQueue(msmqAddress, identity, settings.UseTransactionalQueues);
             }
-            var msmqAddress = MsmqAddress.Parse(address);
-            var queuePath = msmqAddress.PathWithoutPrefix;
 
-            if (msmqAddress.IsRemote)
+            if (queue != null)
             {
-                Logger.Debug("Queue is on remote machine.");
-                Logger.Debug("If this does not succeed (like if the remote machine is disconnected), processing will continue.");
+                using (queue)
+                {
+                    Logger.Debug("Setting queue permissions.");
+                    QueuePermissions.SetPermissionsForQueue(queue, identity);
+                }
             }
+        }
 
-            Logger.Debug($"Checking if queue exists: {address}.");
+        static MessageQueue CreateQueue(MsmqAddress msmqAddress, string account, bool transactional)
```

Comment:
Looking at the way this turned out, and the usage of it above, what if you changed it to be `TryCreateQueue` instead, and return a bool, with an out parameter for the queue instance.